File: 프로그래머스/lv2/12945. 피보나치 수/피보나치 수.java
import java.util.*;
class Solution {
    public int solution(int n) {
        int answer = 0;
        int divide = 1234567;
        int [] arr = new int[n+1];
        arr[0]=0;
        arr[1]=1;
        
        for(int i=2;i<=n;i++){
            arr[i]=arr[i-2]%divide+arr[i-1]%divide;
        }
        
        return arr[n]%divide;
    }
}

File: 프로그래머스/2/12945. 피보나치 수/피보나치 수.py
def solution(n):
    answer = 0
    fibo=[0,1,1,2,3,5]
    
    for i in range(6,n+1):
        fibo.append((fibo[i-2]+fibo[i-1]))
    
    print(fibo[:10])
    return fibo[n]%1234567